File: dataset/Vanilla - Copy/OPENAI_gpt-5-2025-08-07/Task33.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <functional>

int queueTime(const std::vector<int>& customers, int n) {
    if (n <= 0) throw std::invalid_argument("n must be positive");
    std::priority_queue<int, std::vector<int>, std::greater<int>> pq;
    for (int i = 0; i < n; ++i) pq.push(0);
    for (int t : customers) {
        int next = pq.top(); pq.pop();
        pq.push(next + t);
    }
    int total = 0;
    while (!pq.empty()) {
        total = std::max(total, pq.top());
        pq.pop();
    }
    return total;
}

int main() {
    std::vector<std::pair<std::vector<int>, int>> tests = {
        {{5, 3, 4}, 1},
        {{10, 2, 3, 3}, 2},
        {{2, 3, 10}, 2},
        {{}, 1},
        {{1, 2, 3, 4, 5}, 100}
    };
    for (auto& test : tests) {
        std::cout << queueTime(test.first, test.second) << std::endl;
    }
    return 0;
}

File: dataset/Vanilla/OPENAI_gpt-5-2025-08-07/Task33.py
import heapq

def queueTime(customers, n):
    if n <= 0:
        raise ValueError("n must be positive")
    times = [0] * n
    heapq.heapify(times)
    for t in customers:
        shortest = heapq.heappop(times)
        heapq.heappush(times, shortest + t)
    return max(times) if times else 0

if __name__ == "__main__":
    tests = [
        ([5, 3, 4], 1),
        ([10, 2, 3, 3], 2),
        ([2, 3, 10], 2),
        ([], 1),
        ([1, 2, 3, 4, 5], 100),
    ]
    for customers, n in tests:
        print(queueTime(customers, n))